mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -197,3 +197,28 @@ class ChiSqSelector @Since("1.3.0") (
     new ChiSqSelectorModel(indices)
   }
 }
+
+/**
+ * Creates a ChiSquared feature selector by False Positive Rate (FPR) test.
+ * @param alpha the highest p-value for features to be kept
+ */
+@Since("2.1.0")
+class ChiSqSelectorByFpr @Since("2.1.0") (
+  @Since("2.1.0") val alpha: Double) extends Serializable {
+
+  /**
+   * Returns a ChiSquared feature selector by FPR.
+   *
+   * @param data an `RDD[LabeledPoint]` containing the labeled dataset with categorical features.
+   *             Real-valued features will be treated as categorical for each distinct value.
+   *             Apply feature discretizer before using this function.
+   */
+  @Since("2.1.0")
+  def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
+    val indices = Statistics.chiSqTest(data)
+      .zipWithIndex.filter { case (res, _) => res.pValue < alpha }
```

Comment:
Yeah, what I should really say is: why does the model want sorted indices? it can sort them if needed internally. It's consuming just a set of indices, conceptually.

This, sort of, leads to maybe a better proposal. Why not retain the p values in the model by not keeping only the indices? Then, the model itself can be configured to do different types of selection. This would avoid having to make a second (or third or fourth) type of chi-squared selector. In fact all of them are doing the same selection under the hood, so it doesn't really make sense to make new selectors.

What's differing here is how the model uses the chi-squared fit to do feature selection, so I wonder if we can just push the change there? it may make this easier to implement in `.ml` too.